### ISSUE TITLE
Popup's close button should have an accessible name and `role="button"`

### ISSUE THREAD

**Issue by Malvoz** [Leaflet/Leaflet#7399]
## Motivation

The tables below provide screen capture of screen readers' output with the current markup of close buttons for popups, and also of improved markup.

The "improved" markup for comparison uses `aria-label="Close popup"` and `role="button"` on `leaflet-popup-close-button`.

<table>
  <tr>
    <th width="120"><a href="https://chrome.google.com/webstore/detail/screen-reader/kgejglhpjiefppelpmljglcjbhoiplfn?hl=en">ChromeVox</a></th>
    <th>Screen capture</th>
    <th>Screen reader output</th>
  </tr>
  <tr>
    <td>Current</td>
    <td width="300"><video src="https://user-images.githubusercontent.com/26493779/103791768-2349c480-5043-11eb-80a3-3f9b9ae8238e.mp4"></video>
</td>
    <td><q>Multiplication, <i>internal link</i></q></td>
  </tr>
  <tr>
    <td>Improved</td>
    <td width="300"><video src="https://user-images.githubusercontent.com/26493779/103791869-41afc000-5043-11eb-8d91-526cd48cbfc2.mp4"></video></td>
    <td><q>Close popup, <i>button</i></q></td>
  </tr>
</table>

<table>
  <tr>
    <th width="120"><a href="https://www.nvaccess.org/">NVDA</a></th>
    <th>Screen capture</th>
    <th>Screen reader output</th>
  </tr>
  <tr>
    <td>Current</td>
    <td width="300"><video src="https://user-images.githubusercontent.com/26493779/103792111-8b98a600-5043-11eb-8bb9-05280a78fc39.mp4"></video></td>
    <td><q><i>Frame clickable</i>, times, <i>link</i></q></td>
  </tr>
  <tr>
    <td>Improved</td>
    <td width="300"><video src="https://user-images.githubusercontent.com/26493779/103792127-905d5a00-5043-11eb-9800-0e244d043aaf.mp4"></video></td>
    <td><q><i>Clickable</i>, close popup, <i>button</i></q></td>
  </tr>
</table>

## Proposed solution

Set `role="button"` and `aria-label="Close"` (or "Close popup") on the popup's close button:

https://github.com/Leaflet/Leaflet/blob/3466cbba34351ee0d7477a83eb0215cc0e6b454b/src/layer/Popup.js#L192

## Additional context

In a related PR (https://github.com/Leaflet/Leaflet/pull/7079/) `aria-label="Close"` is used, but there's no context for _what_ is to be closed as there is no semantic structure of a "popup", so perhaps it's better to use `aria-label="Close popup"` instead.
